zephyr/sof_syscall.c

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: BSD-3-Clause
```

Comment:
I think we are going to have scheduler, library and llext syscalls so probably best to have a syscall directory here and a file for each subsystem.